### PR TITLE
base: Check downloaded jfrog sha256 sum

### PR DESCRIPTION
**Pull request by leandro-lucarella-sociomantic**
For this we also install the handy tool jq to query json data.

### CODE REVIEW

File: docker/base/jfrog [hunk 1]
```diff
@@ -1,7 +1,29 @@
...
-chmod a+x /usr/local/bin/jfrog
+# Get sha256 sum of the file
+json=$(set -x; curl -#LkX GET "$url/packages/$subj/$repo/$pkg/versions/$ver/files")
+# From the received array, select the object with the specific "path"
+# attribute, and from that object get the "sha256" attribute
+sha256=$(echo "$json" | jq -r ".[] | select(.path == \"$ver/$pkg/$file\") |
+		.sha256")
+
+# Get the file
+(set -x; curl -#LkX GET "$url/$subj/$repo/download_file?file_path=$ver/$pkg/$file" > "$dst")
```
Comment (nemanja-boric-sociomantic) [inline docker/base/jfrog]: Is this commented on purpose?

File: docker/base/jfrog [hunk 2]
```diff
@@ -1,7 +1,29 @@
...
+pkg="jfrog-cli-linux-amd64"
+ver="${1:-$VERSION_JFROG}"
+file="jfrog"
+#dst="/usr/local/bin/$file"
+dst="/tmp/$file"
 
-curl -XGET -L -k "https://api.bintray.com/content/jfrog/jfrog-cli-go/${VERSION_JFROG}/jfrog-cli-linux-amd64/jfrog?bt_package=jfrog-cli-linux-amd64" > /usr/local/bin/jfrog
-chmod a+x /usr/local/bin/jfrog
+# Get sha256 sum of the file
+json=$(set -x; curl -#LkX GET "$url/packages/$subj/$repo/$pkg/versions/$ver/files")
```
Comment (nemanja-boric-sociomantic) [inline docker/base/jfrog]: Is this commented on purpose?

File: docker/base/jfrog (hunk 1 above)
Comment (leandro-lucarella-sociomantic) [reply docker/base/jfrog]: No, that's just a failed `sh` syntax highlighter, VIM has the same problem).

**Comment by leandro-lucarella-sociomantic** [comment]
sha256 sums don't match though. Not for 1.8.0 nor for 1.9.0.

File: docker/base/jfrog [hunk 3]
```diff
@@ -1,7 +1,29 @@
...
+set -eu
 
-VERSION_JFROG=${1:-$VERSION_JFROG}
+url="https://api.bintray.com"
+subj="jfrog"
+repo="jfrog-cli-go"
+pkg="jfrog-cli-linux-amd64"
+ver="${1:-$VERSION_JFROG}"
+file="jfrog"
+#dst="/usr/local/bin/$file"
```
Comment (nemanja-boric-sociomantic) [inline docker/base/jfrog]: looks like a leftover from a development?

**Comment by leandro-lucarella-sociomantic** [comment]
```
$ dst=/tmp/jfrog docker/base/jfrog 1.8.0
+ curl -#LkX GET https://api.bintray.com/packages/jfrog/jfrog-cli-go/jfrog-cli-linux-amd64/versions/1.8.0/files
######################################################################## 100.0%
+ curl -#LkX GET https://api.bintray.com/jfrog/jfrog-cli-go/download_file?file_path=1.8.0/jfrog-cli-linux-amd64/jfrog
######################################################################## 100.0%
+ test 455a3b10d731c735848acd2c31d91ff4c414e8fdae554553d2a14b3cc13cc621 = 2193fef2d209d7ad2c20bd195daf17c58740b25dd10d1c5e3710ba48dce858a5

$ dst=/tmp/jfrog docker/base/jfrog 1.9.0
+ curl -#LkX GET https://api.bintray.com/packages/jfrog/jfrog-cli-go/jfrog-cli-linux-amd64/versions/1.9.0/files
######################################################################## 100.0%
+ curl -#LkX GET https://api.bintray.com/jfrog/jfrog-cli-go/download_file?file_path=1.9.0/jfrog-cli-linux-amd64/jfrog
######################################################################## 100.0%
+ test 455a3b10d731c735848acd2c31d91ff4c414e8fdae554553d2a14b3cc13cc621 = 4ec5550aee278386ab9ebbcdf95fb14b2ce82a0979d66cba92fdcea97f9aaa0f
```

File: docker/base/jfrog (hunk 1 above)
Comment (nemanja-boric-sociomantic) [reply docker/base/jfrog]: Ha!

File: docker/base/jfrog (hunk 3 above)
Comment (leandro-lucarella-sociomantic) [reply docker/base/jfrog]: Ah, yeah. I will remove that.

**Comment by leandro-lucarella-sociomantic** [comment]
I will report.

**Comment by leandro-lucarella-sociomantic** [comment]
Updated.

**Comment by leandro-lucarella-sociomantic** [comment]
Using the sha1 works, but it turns out the downloaded 1.8.0 and the 1.9.0 binaries are the same (according to the sha sum). When using the sha1 even the uploaded binaries look the same. This is not precisely instilling trust... :worried: 

https://github.com/JFrogDev/jfrog-cli-go/issues/62

**Comment by leandro-lucarella-sociomantic** [comment]
Blocked until upstream issue gets resolved. Also after we merge the travis stuff we will get an error in the PR until is fixed :)

**Comment by leandro-lucarella-sociomantic** [comment]
I can't believe I will say this, but I'm happy that it's finally red :P

**Comment by leandro-lucarella-sociomantic** [comment]
Oh, this is because I removed the branch it was based on. I will re-open.

**Comment by leandro-lucarella-sociomantic** [comment]
https://github.com/sociomantic-tsunami/cachalot/pull/13

**Comment by leandro-lucarella-sociomantic** [comment]
OK, that URL needs authentication, so I was just getting a json response. Update coming soon.

**Comment by leandro-lucarella-sociomantic** [comment]
Arrgh, not this one, #13 :D

**Pull request closed without merging**